matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/job/SyncThread.kt

```diff
@@ -174,6 +179,13 @@ internal class SyncThread @Inject constructor(private val syncTask: SyncTask,
         networkConnectivityChecker.unregister(this)
     }
 
+    private fun observeActiveCalls() {
+        syncScope.launch(Dispatchers.Main) {
+            activeCallHandler.getActiveCallsLiveData().removeObserver(activeCallListObserver)
```

Comment:
If you accidentally try to add the same observer then the application will crash and there is no easy way to check whether the observer already added or not. I checked the implementation of the removeObserver method. It is just checking a map and does nothing if can't find it. So I don't think any harm since it is O(1).

Reply:
Why would we accidentally try to add the same observer?

Reply:
This code should be moved at the end of the run() method then

Reply:
Fixed the registration flow of the observer.